### PR TITLE
W-16440854-Add-searchFieldsAsType-Option-CP

### PR DESCRIPTION
**Pull request by Crispy-Salesforce**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: modules/ROOT/pages/apikit-odatav4-extension-reference.adoc [hunk 1]
```diff
@@ -226,39 +226,40 @@ Default configuration
...
-| Streaming Strategy |
+| Config Ref | ConfigurationProvider | Name of the configuration used to execute this component |  | Yes
+| Configuration | String | Name of the configuration to use | | Yes
+| Database Schema | String | Database schema name to which the entity set belongs. If not empty, it's prefixed to the table name in the `WHERE` clause |  | No
+| Entity Set Name | String | Table name in the database for the requested entity set |  | Yes
+| Entity Type Fields | Array of String | List of property fields from the Entity type |  | Yes
+| Entity Type Keys | Object | Map of entity key names and values |  | Yes
+| Entity Type Name | String | Entity Data Model specifies the entity type name |  | Yes
+| Page Size | Number |  |  | No
+| searchFieldsAsType | Object | Select if the DB columns is typecasted to TEXT when using the `$search` operation | | No
```
Comment (aby-philip-sf) [inline modules/ROOT/pages/apikit-odatav4-extension-reference.adoc]: Not sure if this description is auto-generated. The description of this field in code is `Option to typecast fields for search transformation`

The description is not accurate. Consider rephrasing as
Data type to which fields are type-casted for `$search` operation. If empty, all fields are typecast as TEXT by default
cc @ruturajgujar

Comment (aby-philip-sf) [reply modules/ROOT/pages/apikit-odatav4-extension-reference.adoc]: Thanks to @ruturajgujar, please update the description as follows

Data type to which fields are type-casted for $search operation. If empty, no type-casting is performed.

Comment (ruturajgujar) [reply modules/ROOT/pages/apikit-odatav4-extension-reference.adoc]: @Crispy-Salesforce I have updated the field description as per Aby's suggestion. Can you please update the reference doc file?

Comment (Crispy-Salesforce) [reply modules/ROOT/pages/apikit-odatav4-extension-reference.adoc]: Hi, team! Thanks for your feedback!

**Review by aby-philip-sf — APPROVED** [review]

**Review by failup — APPROVED** [review]

**Pull request merged**